test/parametrized_tester.c

```diff
@@ -0,0 +1,194 @@
+/* parametrized_tester.c - Test deflate() and inflate() on a chunk of the
+ * input data file, combining many deflate() settings to improve testing coverage.
+ */
+#include "zbuild.h"
+#ifdef ZLIB_COMPAT
+#  include "zlib.h"
+#else
+#  include "zlib-ng.h"
+#endif
+
+#include <stdio.h>
+
+/*
+ * WITH_EXTRA_TESTS results in 9450 permutations per input file.
+ * Normal mode results in 540.
+ *
+ * WITH_EXTRA_TESTS also tests a bigger chunk of the files.
+ */
+
+#ifdef WITH_EXTRA_TESTS
+const int valid_wbits[] = {
+    -9, -10, -11, -12, -13, -14, -15,
+     9,  10,  11,  12,  13,  14,  15,
+# ifdef WITH_GZFILEOP
+    25,  26,  27,  28,  29,  30,  31,
+# endif
+};
+
+const int valid_compression_levels[] = {
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9
+};
+
+const int valid_mem_levels[] = {
+    1, 2, 3, 4, 5, 6, 7, 8, 9,
+};
+
+#define MAX_FILESIZE_KB 4096
+
+# ifdef WITH_GZFILEOP
+#  define NUM_W 21
+# else
+#  define NUM_W 14
+# endif
+# define NUM_C 10
+# define NUM_M 9
+
+#else // !WITH_EXTRA_TESTS
+
+const int valid_wbits[] = {
+    -9, -15,
+     9,  15,
+# ifdef WITH_GZFILEOP
+    25,  31,
+# endif
+};
+
+const int valid_compression_levels[] = {
+    0, 1, 2, 4, 6, 9
+};
+
+const int valid_mem_levels[] = {
+    1, 8, 9,
+};
+
+#define MAX_FILESIZE_KB 128
+
+# ifdef WITH_GZFILEOP
+#  define NUM_W 6
+# else
+#  define NUM_W 4
+# endif
+# define NUM_C 6
+# define NUM_M 3
+#endif
+
+#define MAX_FILESIZE MAX_FILESIZE_KB * 1024
+#define MAX_OUTSIZE MAX_FILESIZE_KB * (1024 + 128)
+
+const int valid_strategies[] = {
+    Z_FILTERED,
+    Z_HUFFMAN_ONLY,
+    Z_RLE,
+    Z_FIXED,
+    Z_DEFAULT_STRATEGY,
+};
+
+int main (int argc, char *argv[])  {
+    PREFIX3(stream) zst_deflate;
```

Comment:
Can we use `d_stream` and `c_stream` like elsewhere in our example code?

Reply:
Changed.